[WFLY-15849] OpenTelemetry QS under OpenShift

emmartins opened the pull request:
Clean up YAML
Add instructions for deploying otel collector on OS

Issue: [WFLY-15849](https://issues.redhat.com/browse/WFLY-15849)

emmartins commented:
Superseeds #870

kabir commented:
@emmartins For some reason the opentelemetry tests are not getting picked up. See https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/wildfly_quickstart/872/pull-ci-wildfly-quickstart-main-wildfly-quickstart-e2e/1752305685376274432/artifacts/wildfly-quickstart-e2e/quickstart-build/build-log.txt

Right at the end
````
Parsed test directories, and determined tests should be run for the following directories:
	
Running tests...
All tests run in 0m0s.
\033[0;32mAll tests passed!\033[0m

emmartins commented:
AFAIK That will be fixed by another PR.

On Tue, 30 Jan 2024 at 13:17, Kabir Khan ***@***.***> wrote:

> @emmartins <https://github.com/emmartins> For some reason the
> opentelemetry tests are not getting picked up. See
> https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/wildfly_quickstart/872/pull-ci-wildfly-quickstart-main-wildfly-quickstart-e2e/1752305685376274432/artifacts/wildfly-quickstart-e2e/quickstart-build/build-log.txt
>
> Right at the end
>
> Parsed test directories, and determined tests should be run for the following directories:
> 	
> Running tests...
> All tests run in 0m0s.
> \033[0;32mAll tests passed!\033[0m
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/wildfly/quickstart/pull/872#issuecomment-1916830121>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAEDY6HWZCCU4FG7J4WWBOLYRDXHFAVCNFSM6AAAAABCRDS3OCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMJWHAZTAMJSGE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

kabir commented:
It might be because the GH Action workflow is called `quickstart_opentelemetry_ci.yml` while the quickstart folder is called `opentelemetry-tracing`, which would cause it to miss this check https://github.com/wildfly/quickstart/blob/main/.ci/openshift-ci/build-root/scripts/openshift-test-runner.sh#L151-L153

Either the `quickstart_opentelemetry_ci.yml` file needs to be called  `quickstart_opentelemetry-tracing_ci.yml`, or the quickstart folder needs to be renamed to `opentelemetry`

jasondlee commented:
> Either the `quickstart_opentelemetry_ci.yml` file needs to be called `quickstart_opentelemetry-tracing_ci.yml`, or the quickstart folder needs to be renamed to `opentelemetry`

I can address that issue here: https://github.com/wildfly/quickstart/pull/871

emmartins commented:
> It might be because the GH Action workflow is called `quickstart_opentelemetry_ci.yml` while the quickstart folder is called `opentelemetry-tracing`, which would cause it to miss this check https://github.com/wildfly/quickstart/blob/main/.ci/openshift-ci/build-root/scripts/openshift-test-runner.sh#L151-L153
> 
> Either the `quickstart_opentelemetry_ci.yml` file needs to be called `quickstart_opentelemetry-tracing_ci.yml`, or the quickstart folder needs to be renamed to `opentelemetry`

fixed, added functions override and remove test exclusion

jasondlee reviewed (approved):
Looks good to me. I'll close my PR in favor of this one.

kabir reviewed (approved):
I've checked the OpenShift tests, and they are running

kstekovi commented:
The OpenShift guide have unorganized steps compare to bootabale jar. 
bootable-jar:
- build
- start app
- start integration tests
- stop 

OpenShift:
- prerequisites (start the opentelemetry) 
- deploy QS
- uninstall
- start integration tests

emmartins commented:
That issue of undeploy before testing affects all QS, it is something we
need to address separately, please ignore on this PR.

On Thu, 1 Feb 2024 at 09:58, Krystof Stekovic ***@***.***>
wrote:

> The OpenShift guide have unorganized steps compare to bootabale jar.
> bootable-jar:
>
>    - build
>    - start app
>    - start integration tests
>    - stop
>
> OpenShift:
>
>    - prerequisites (start the opentelemetry)
>    - deploy QS
>    - uninstall
>    - start integration tests
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/wildfly/quickstart/pull/872#issuecomment-1920942826>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAEDY6DBRVNGR5R5SLAUMH3YRNRKXAVCNFSM6AAAAABCRDS3OCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMYTSMRQHE2DEOBSGY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

kstekovi commented on .ci/openshift-ci/build-root/scripts/qs-overrides/opentelemetry-tracing/overridable-functions.sh:
```diff
@@ -0,0 +1,9 @@
+
+function cleanPrerequisites() {
+  echo "Deleting all OpenTelemetry Collector resources"
+  oc delete -f charts/opentelemetry-collector.yaml
```
I think this step should as additional step in undeploy

emmartins replied:
The proper solution for this requires an additional extension point for shared-docs, similar to helm-install-prerequisites, and that will affect all QS so I have created https://issues.redhat.com/browse/WFLY-18986 to deliver it (and more general shared improvements). For now the uninstall is referenced as a note at https://github.com/wildfly/quickstart/pull/872/files#diff-437f0901bdbc94e992d4b6b75f355240a557c4879683f47d27dad1c51f27dc4aR19

emmartins commented:
I did think of that, but we will need to add another extension point on the
shared-docs, similar to what we have for extra pre requisites install. Due
to tight schedule for now I opted for a note on the pre requisites install
content, but definitely in the future I would like to add such undeploy
extension point and have mp reactive, micrometer and opentelemetry QS to
provide an impl for that. If you agree of course, your call, you have the
right to request that now if you want :-)

On Thu, 1 Feb 2024 at 10:02, Krystof Stekovic ***@***.***>
wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In
> .ci/openshift-ci/build-root/scripts/qs-overrides/opentelemetry-tracing/overridable-functions.sh
> <https://github.com/wildfly/quickstart/pull/872#discussion_r1474161382>:
>
> > @@ -0,0 +1,9 @@
> +function runPostHelmInstallCommands() {
> +  echo "Applying all OpenTelemetry Collector resources"
> +  oc apply -f charts/opentelemetry-collector.yaml
> +}
> +
> +function cleanPrerequisites() {
> +  echo "Deleting all OpenTelemetry Collector resources"
> +  oc delete -f charts/opentelemetry-collector.yaml
>
> I think this step should as additional step in undeploy
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/wildfly/quickstart/pull/872#pullrequestreview-1856069729>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAEDY6ECDHVNBEZT5XFT7OTYRNR3JAVCNFSM6AAAAABCRDS3OCVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTQNJWGA3DSNZSHE>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

kstekovi commented:
It seem to be similar issues with steps organization. So I think this could be fixed in that scope. https://github.com/wildfly/quickstart/pull/872#issuecomment-1920947662

> I did think of that, but we will need to add another extension point on the shared-docs, similar to what we have for extra pre requisites install. Due to tight schedule for now I opted for a note on the pre requisites install content, but definitely in the future I would like to add such undeploy extension point and have mp reactive, micrometer and opentelemetry QS to provide an impl for that. If you agree of course, your call, you have the right to request that now if you want :-)
> […](#)
> On Thu, 1 Feb 2024 at 10:02, Krystof Stekovic ***@***.***> wrote: ***@***.**** commented on this pull request. ------------------------------ In .ci/openshift-ci/build-root/scripts/qs-overrides/opentelemetry-tracing/overridable-functions.sh <[#872 (comment)](https://github.com/wildfly/quickstart/pull/872#discussion_r1474161382)>: > @@ -0,0 +1,9 @@ +function runPostHelmInstallCommands() { + echo "Applying all OpenTelemetry Collector resources" + oc apply -f charts/opentelemetry-collector.yaml +} + +function cleanPrerequisites() { + echo "Deleting all OpenTelemetry Collector resources" + oc delete -f charts/opentelemetry-collector.yaml I think this step should as additional step in undeploy — Reply to this email directly, view it on GitHub <[#872 (review)](https://github.com/wildfly/quickstart/pull/872#pullrequestreview-1856069729)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AAEDY6ECDHVNBEZT5XFT7OTYRNR3JAVCNFSM6AAAAABCRDS3OCVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMYTQNJWGA3DSNZSHE> . You are receiving this because you were mentioned.Message ID: ***@***.***>

kstekovi commented:
OpenShift steps works as is expected. Ready to merge. Thanks.

emmartins commented:
thank you @kstekovi
